src/api/services/NotificationAPI.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * NotificationAPI Services
+ * Provide services related to Notification (Avvisatura) to communicate with PagoPaAPI
+ */
+
+import * as express from "express";
+import fetch from "node-fetch";
+import * as uuid from "uuid";
+import { PagoPaConfig } from "../../Configuration";
+import { NotificationSubscriptionRequestType } from "../../enums/NotificationSubscriptionType";
+import { FiscalCode } from "../../types/FiscalCode";
+import { NotificationSubscriptionResponse } from "../types/NotificationSubscriptionResponse";
+
+// Update subscription (Activation or Deactivation) to Notification Service for a fiscalCode
+export function updateSubscription(
+  fiscalCode: FiscalCode,
+  requestType: NotificationSubscriptionRequestType,
+  res: express.Response,
+  pagoPaConfig: PagoPaConfig,
+  errorCallback: (res: express.Response, errorMsg: string) => void,
+  successCallback: (
+    res: express.Response,
+    notificationSubscriptionResponse: NotificationSubscriptionResponse
+  ) => void
+): void {
+  const url = `${pagoPaConfig.HOST}:${pagoPaConfig.PORT}${
+    pagoPaConfig.SERVICES.NOTIFICATION_UPDATE_SUBSCRIPTION
+  }`;
+  const body = {
+    timestamp: Date.now(),
+    requestId: uuid.v1(),
+    operation:
+      requestType === NotificationSubscriptionRequestType.ACTIVATION
+        ? "A" // Activation
```

Comment:
instead of comments you may consider to assign these hardcoded strings to some constant or enum (and change the type from NonEmptyString)